compiler/testData/codegen/bytecodeText/assert/jvmNestedClass.kt

```diff
@@ -0,0 +1,16 @@
+// IGNORE_BACKEND: JVM
+// KOTLIN_CONFIGURATION_FLAGS: ASSERTIONS_MODE=jvm
+
+class Outer {
+    class Inner {
+        fun f() { assert(true) }
+    }
+}
+
+// We set the assertion status based on top-level classes.
+// 0 LDC LOuter\$Inner;.class
+// Outer\$Inner.<clinit>:
+// 1 LDC LOuter;.class\s*INVOKEVIRTUAL java/lang/Class.desiredAssertionStatus \(\)Z
+// 1 PUTSTATIC Outer\$Inner.\$assertionsDisabled : Z
+// Outer\$Inner.f:
+// 1 GETSTATIC Outer\$Inner.\$assertionsDisabled
```

Comment:
Please also add:
// 1 desiredAssertionStatus
// 1 public final static Z \$assertionsDisabled

Reply:
Done!

Reply:
@sfs Sorry, but actually in java all '$assertionsDisabled' fields are package local and synthetic:
```
final static synthetic Z $assertionsDisabled
```

Reply:
Thanks, that's good to know. I've changed the test.

Currently the IR backend has problems with field visibility and this is just one more test case which we'll have to fix down the line...